docs/ARCHITECTURE.md

```diff
@@ -240,24 +344,41 @@ direction LR
             +usingOutputMime(string $mimeType) self
             +usingOutputSchema(array< string, mixed > $schema) self
             +usingOutputModalities(...AiModality $modalities) self
+            +asArrayResponse(?array< string, mixed > $schema) self
```

Comment:
I think we should remove this - the response isn't necessarily an array in the sense of JSON - it'll often be a JSON object. In PHP that's still an (associative) array, but still IMO this is a confusing name. Since there's already `asJsonResponse`, I don't see a need for this alternative anyway.